### PR TITLE
Reworking material stacks and material spawning.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
In an effort to make material sheet behavior less type-dependent I've removed all non-generic material sheets and swapped out direct spawning of sheets to use `material.place_sheet()`. I've also moved sheet single/plural names onto the sheets themselves and replaced stacktype on the material with a `default_solid_form` var.

:cl:
tweak: Some material stack items and messages around them may have changed due to a backend refactor of materials.
/:cl:

### CODE REVIEW

File: code/game/machinery/camera/camera_assembly.dm [hunk 1]
```diff
@@ -10,7 +10,11 @@
 	matter = list(/decl/material/solid/glass = MATTER_AMOUNT_REINFORCEMENT)
 
 	//	Motion, EMP-Proof, X-Ray
-	var/list/obj/item/possible_upgrades = list(/obj/item/assembly/prox_sensor, /obj/item/stack/material/osmium, /obj/item/stock_parts/scanning_module)
+	var/list/obj/item/possible_upgrades = list(
+		/obj/item/assembly/prox_sensor,
+		/obj/item/robot_parts/robot_component/armour,
```
Comment (MistakeNot4892) [inline code/game/machinery/camera/camera_assembly.dm]: TODO: maybe add an EMP-hardened version of this.

File: code/game/objects/structures/rubble.dm [hunk 2]
```diff
@@ -8,7 +8,11 @@
 	anchored = 1
 	maxhealth = 50
 
-	var/list/loot = list(/obj/item/cell,/obj/item/stack/material/iron,/obj/item/stack/material/rods)
+	var/list/loot = list(
```
Comment (MistakeNot4892) [inline code/game/objects/structures/rubble.dm:11]: TODO: work the spawners into the icon gen proc for these somehow.

File: code/__defines/research.dm [hunk 3]
```diff
@@ -8,6 +8,8 @@
 #define REAGENT_UNITS_PER_MATERIAL_SHEET 20
 #define REAGENT_UNITS_PER_GAS_MOLE 10
 
+#define MOLES_PER_SHEET 50
```
Comment (MistakeNot4892) [inline code/__defines/research.dm]: This is likely too high, it was copied from the oxygen recycler.

File: code/game/objects/items/stacks/rods.dm [hunk 4]
```diff
@@ -66,7 +61,7 @@
 			return
 
 		if(WT.remove_fuel(0,user))
-			var/obj/item/stack/material/steel/new_item = new(usr.loc)
+			var/obj/item/stack/material/new_item = new(usr.loc, 1, material?.type || /decl/material/solid/metal/steel)
```
Comment (comma) [inline code/game/objects/items/stacks/rods.dm]: doesn't this need to use fancy shmancy new create object proc?

File: code/game/objects/items/weapons/material/shards.dm [hunk 5]
```diff
@@ -18,6 +18,7 @@
 	applies_material_name = TRUE
 	unbreakable = 1 //It's already broken.
 	item_flags = ITEM_FLAG_CAN_HIDE_IN_SHOES
+	atom_flags = 0
```
Comment (comma) [inline code/game/objects/items/weapons/material/shards.dm]: why dis

Comment (MistakeNot4892) [reply code/game/objects/items/weapons/material/shards.dm]: ATOM_FLAG_NO_TEMPERATURE_CHANGE or whatever the macro is prevents temperature stuff from being called and is set by default.

File: code/game/objects/items/stacks/rods.dm (hunk 4 above)
Comment (MistakeNot4892) [reply code/game/objects/items/stacks/rods.dm]: Yeah I must have missed it.

**Comment by MistakeNot4892** [comment]
Depends on #1535.

**Comment by MistakeNot4892** [comment]
If anyone has any insight into why my changes are making cargo costs change, please let me know because I'm stumped.

File: code/modules/hydroponics/grown.dm [hunk 6]
```diff
@@ -147,6 +147,16 @@
 	if(seed) 
 		seed.thrown_at(src,hit_atom)
 
+var/list/_wood_materials = list(
```
Comment (comma) [inline code/modules/hydroponics/grown.dm]: can't this just be istype(/decl/material/solid/wood)?

Comment (MistakeNot4892) [reply code/modules/hydroponics/grown.dm]: You'd have to iterate the entire chems list and it would include stuff like holographic wood.

**Comment by MistakeNot4892** [comment]
As far as I can tell this is ready to go now.

File: code/modules/power/fusion/kinetic_harvester.dm [hunk 7]
```diff
@@ -1,3 +1,4 @@
+// This machine is COMPLETELY BROKEN due to material and reagent reworks, TODO: total overhaul
```
Comment (comma) [inline code/modules/power/fusion/kinetic_harvester.dm:1]: is it tho, i thought you fixed it in another PR

Comment (MistakeNot4892) [reply code/modules/power/fusion/kinetic_harvester.dm:1]: It's not fixed on dev yet.

**Review by comma — DISMISSED** [review]

**Review by afterthought2 — DISMISSED** [review]

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**